packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/textEditorParts/mask.tests.js

```diff
@@ -2334,6 +2335,42 @@ QUnit.module('validation', {}, () => {
         textEditor.option('value', 'f');
         assert.notOk(textEditor.option('isValid'), 'mask with an invalid value should be invalid');
     });
+
+    QUnit.test('The validation error should be null after removing the mask (T1214604)', function(assert) {
+        const maskInvalidMessage = 'mask error';
+        const $textEditor = $('#texteditor').dxTextEditor({
+            mask: '000000/0009',
+            maskInvalidMessage,
+        });
+        const textEditor = $textEditor.dxTextEditor('instance');
+
+        textEditor.option({ value: '123' });
+
+        assert.strictEqual(textEditor.option('validationError').message, maskInvalidMessage, 'validation message was added');
+
+        textEditor.option({ mask: '' });
+
+        assert.strictEqual(textEditor.option('validationError'), null);
+    });
+
+    QUnit.test('The validation status in Validator should be valid after removing the mask (T1214604)', function(assert) {
+        const $textEditor = $('#texteditor').dxTextEditor({
+            mask: '000000/0009',
+        }).dxValidator({
+            validationRules: [{ type: 'required' }],
+        });
+
+        const textEditor = $textEditor.dxTextEditor('instance');
+        const validator = $textEditor.dxValidator('instance');
+
+        textEditor.option({ value: '123' });
+
+        assert.strictEqual(validator.option('validationStatus'), 'invalid');
+
+        textEditor.option({ mask: '' });
+
+        assert.strictEqual(validator.option('validationStatus'), 'valid');
```

Comment:
I mean checking that custom validation error is kept after mask restoring.
Like:
```javascript
validationRules: [{
    type: 'custom',
    callback: () => false,
    message: 'custom error'
}],
...
checking that we have an error with "custom error" message
```

Reply:
[9ea75f939d6413748c4ea3932566aefc4a6158ec]